File: QE_FST_Selenium_java/Activity12_waits2.java
package demos;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
//import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity12_waits2 {
    public static void main(String[] args) throws InterruptedException {
        WebDriver driver=new FirefoxDriver();
        driver.get("https://training-support.net/webelements/dynamic-content");
        System.out.println(driver.getTitle());
        Thread.sleep(2000);
         WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(20));
By messageLocator = By.id("word");
           
wait.until(ExpectedConditions.textToBePresentInElementLocated(
        messageLocator, "release"
));
            String text=driver.findElement(messageLocator).getText();
            System.out.println("The result is:"+text);
            driver.quit();


    }
}

File: QE_FST_Selenium_python/Activity12_waits2.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

with webdriver.Firefox() as driver:
    driver.get("https://training-support.net/webelements/dynamic-content")
    print(driver.title)
    wait = WebDriverWait(driver, 30)
    clickButton=driver.find_element(By.ID,"genButton")
    clickButton.click()
    words=(By.ID,"word")
    wait.until(EC.text_to_be_present_in_element(words, "release"))
    print(driver.find_element(*words).text)
